common/namer/namer.go

```diff
@@ -55,3 +61,22 @@ func (n Namer) DisplayName(parts ...pulumi.StringInput) pulumi.StringInput {
 		return n.ResourceName(strArgs...)
 	}).(pulumi.StringOutput)
 }
+
+// HashedName return the md5 32-chars hex pulumi.StringInput of `parts` prefixed
+// with the namer prefix
+func (n Namer) HashedName(parts ...pulumi.StringInput) pulumi.StringInput {
+	return n.DisplayName(parts...).ToStringOutput().ApplyT(func(name string) string {
+		hmd5 := md5.Sum([]byte(name))
+		return hex.EncodeToString(hmd5[:])
+	}).(pulumi.StringOutput)
+}
+
+// DisplayHashedName return the first 15 chars of the display name plus the first 16 chars
+// of the md5 32-chars hex pulumi.StringInput of display name
+func (n Namer) DisplayHashedName(parts ...pulumi.StringInput) pulumi.StringInput {
+	return pulumi.All(n.DisplayName(parts...), n.HashedName(parts...)).ApplyT(func(args []interface{}) string {
```

Comment:
As `HashedNamed` is only used in this function, you can inline the code in this function and write something like
```go
return n.DisplayName(parts...).ToStringOutput().ApplyT(func(names ...string) string {
 displayName :=   n.DisplayName(names...)
if len(displayName) <= 32 {
    return displayName 
 }
hmd5 := md5.Sum([]byte(displayName))
hash := hex.EncodeToString(hmd5[:])
 return strings.Join([]string{displayName[:15], hashedName[:16]}, "-")
}
```

Reply:
As per the discussion with @L3n41c we will use this in other spots, wherever AWS is expecting a limited string size. I will add a max size to the `HashDisplayName` function and return the display name if the size is fine, or half of the string and half of the hash if it is too long

Reply:
I'm considering having a `Limiter` out of namer, better the limiter logic doesn't have to know about `DisplayName`. It takes a string and limit it to half of it plus adds a substring of its hash to try to reduce collisions